packages/query/src/index.ts

```diff
@@ -1287,61 +1349,44 @@ const generateQueryHook = async (
               options: query?.options,
               type: QueryType.SUSPENSE_INFINITE,
               queryParam: query?.useInfiniteQueryParam,
+              queryKeyFnName: camel(`get-${operationName}-query-key`),
             },
           ]
         : []),
     ];
 
-    const queryKeyFnName = camel(`get-${operationName}-queryKey`);
-    // Convert "param: Type" to "param?: Type" for queryKey functions
-    // to enable cache invalidation without type assertion
-    const makeParamsOptional = (params: string) => {
-      if (!params) return '';
-      // Handle parameters with default values: "param?: Type = value" -> "param: Type = value" (remove optional marker)
-      // Handle regular parameters: "param: Type" -> "param?: Type"
-      return params.replace(
-        /(\w+)(\?)?:\s*([^=,}]*?)\s*(=\s*[^,}]*)?([,}]|$)/g,
-        (match, paramName, optionalMarker, type, defaultValue, suffix) => {
-          // If parameter has a default value, don't add '?' (it's already effectively optional)
-          if (defaultValue) {
-            return `${paramName}: ${type.trim()}${defaultValue}${suffix}`;
-          }
-          // Otherwise, make it optional
-          return `${paramName}?: ${type.trim()}${suffix}`;
-        },
-      );
-    };
-
-    const queryKeyProps = makeParamsOptional(
-      toObjectString(
-        props.filter((prop) => prop.type !== GetterPropType.HEADER),
-        'implementation',
-      ),
-    );
-
-    const routeString =
-      isVue(outputClient) || override.query.shouldSplitQueryKey
-        ? getRouteAsArray(route) // Note: this is required for reactivity to work, we will lose it if route params are converted into string, only as array they will be tracked // TODO: add tests for this
-        : `\`${route}\``;
-
-    // Note: do not unref() params in Vue - this will make key lose reactivity
-    const queryKeyFn = `${
-      override.query.shouldExportQueryKey ? 'export ' : ''
-    }const ${queryKeyFnName} = (${queryKeyProps}) => {
-    return [${routeString}${queryParams ? ', ...(params ? [params]: [])' : ''}${
-      body.implementation ? `, ${body.implementation}` : ''
-    }] as const;
-    }`;
-
-    implementation += `${!queryKeyMutator ? queryKeyFn : ''}
+    implementation += `
+${
+  !queryKeyMutator
+    ? queries
+        .filter(
```

Comment:
reject all non unique queryKeyFnName for prevent generate duplication functions

Reply:
Does this mean that the issue was already there?
Or has this change meant that there is now a possibility of duplication?

Reply:
After the discussion, if we define this processing, insert a variable definition such as `uniqueQueryKeys` to make the nesting shallowe.

Reply:
Unfortunately, the simple solution didn't work.
We need an array of keys: one for regular queries, the other for infinite ones.
Therefore, we have to iterate through the array of queries to collect the required ones.

If we stick with the ternary solution I suggested above, we'll end up with duplicate keys.

Perhaps I'm approaching the problem incorrectly and there's a simpler way. If so, I'd appreciate any hints.